src/phoenix/server/api/mutations/dataset_mutations.py

```diff
@@ -265,6 +266,88 @@ async def add_examples_to_dataset(
             )
         )
 
+    @strawberry.mutation
+    async def delete_dataset_examples(
+        self, info: Info[Context, None], input: DeleteDatasetExamplesInput
+    ) -> DatasetMutationPayload:
+        example_db_ids = [
+            from_global_id_with_expected_type(global_id, models.DatasetExample.__name__)
+            for global_id in input.example_ids
+        ]
+        # Guard against empty input
+        if not example_db_ids:
+            raise ValueError("Must provide examples to delete")
+        dataset_version_description = (
+            input.dataset_version_description
+            if isinstance(input.dataset_version_description, str)
+            else None
+        )
+        dataset_version_metadata = input.dataset_version_metadata or {}
+        async with info.context.db() as session:
+            # Check if the examples are from a single dataset
+            dataset_ids = (
+                (
+                    await session.execute(
+                        select(models.DatasetExample.dataset_id)
+                        .where(models.DatasetExample.id.in_(example_db_ids))
+                        .distinct()
+                    )
+                )
+                .scalars()
+                .all()
+            )
+            if len(dataset_ids) > 1:
+                raise ValueError("Examples must be from the same dataset")
+            elif not dataset_ids:
+                raise ValueError("Examples not found")
+
+            # Get the dataset and raise an error if it doesn't exist
+            dataset_db_id = dataset_ids[0]
+
+            dataset_version_rowid = await session.scalar(
+                insert(models.DatasetVersion)
+                .values(
+                    dataset_id=dataset_db_id,
+                    description=dataset_version_description,
+                    metadata_=dataset_version_metadata,
+                )
+                .returning(models.DatasetVersion.id)
+            )
+            DatasetExampleRevision = models.DatasetExampleRevision
+            await session.execute(
+                insert(DatasetExampleRevision),
+                [
+                    {
+                        DatasetExampleRevision.dataset_example_id.key: dataset_example_rowid,
+                        DatasetExampleRevision.dataset_version_id.key: dataset_version_rowid,
+                        DatasetExampleRevision.input.key: {},
+                        DatasetExampleRevision.output.key: {},
+                        DatasetExampleRevision.metadata_.key: {},
+                        DatasetExampleRevision.revision_kind.key: "DELETE",
```

Comment:
Need to filter out the examples that are already deleted. Double-deleting would causing the following stat to be miscalculated. https://github.com/Arize-ai/phoenix/blob/7bf16b86fea37a906e56425c684ad73f4534bf79/src/phoenix/db/models.py#L407-L408

Reply:
Ah I see, yeah I was punting on that but that makes sense. Good catch.

Reply:
Aborting the entire thing if the set contains an example that's already deleted.